src/symbol/sprite_atlas.js

```diff
@@ -14,9 +16,11 @@ class AtlasImage {
     }
 }
 
-class SpriteAtlas {
+class SpriteAtlas extends Evented {
```

Comment:
All fair points! 

I lost about a week trying to refactor, simplify, and document our whole style image subsystem and eventually decided that the minimal patch to add this feature was worth executing. 

I'll push another commit with some targeted renames & comments to better explain `SpriteAtlas` until we get around to a bigger cleanup.